sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/CBDE/CbdeHandler.cs

```diff
@@ -35,6 +35,43 @@
 
 namespace SonarAnalyzer.Rules.CSharp
 {
+    public class ThreadCpuStopWatch
+    {
+        private double totalMsStart;
+        private readonly ProcessThread currentProcessThread;
+        public long ElapsedMilliseconds
+        {
+            get { return (long)(currentProcessThread.TotalProcessorTime.TotalMilliseconds - totalMsStart); }
+        }
+        public static ProcessThread GetCurrentProcessThread()
+        {
+            var currentId = AppDomain.GetCurrentThreadId();
+            foreach (ProcessThread p in Process.GetCurrentProcess().Threads)
+            {
+                if (p.Id == currentId)
+                {
+                    return p;
+                }
+            }
+            return null;
+        }
+        public void Reset()
+        {
+            totalMsStart = currentProcessThread.TotalProcessorTime.TotalMilliseconds;
+        }
+        public ThreadCpuStopWatch()
+        {
+            currentProcessThread = GetCurrentProcessThread();
+            Reset();
+        }
+        public static ThreadCpuStopWatch StartNew()
+        {
+            var instance = new ThreadCpuStopWatch();
+            instance.Reset();
```

Comment:
Why would you need that? Isn't it already done in the constructor?
And why do you need this factory function, isn't the constructor enough?

Reply:
I left a comment. I am copying the interface of the class Stopwatch we are using for elapsed real time.